scripts/zones/West_Ronfaure/npcs/Vilatroire.lua

```diff
@@ -5,29 +5,200 @@
-        player:startEvent(135) -- Starts first quest - 6 members same alliance
-    elseif intermedTmwrk == QUEST_ACCEPTED then
-        player:startEvent(134) -- You don't have the requirements to finish
+    -- get players fame for this quest region
+    local sandyFame = player:getFameLevel(SANDORIA)
+
+    -- get quest statuses
+    local questIntroToTeamwork = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.INTRODUCTION_TO_TEAMWORK)
+    local questIntermediateTeamwork = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.INTERMEDIATE_TEAMWORK)
+    local questAdvancedTeamwork = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.ADVANCED_TEAMWORK)
+
+    if questIntroToTeamwork == QUEST_AVAILABLE and sandyFame >= 2 then
+        -- start the event that starts the quest
+         player:startEvent(135)
+    elseif questIntroToTeamwork == QUEST_AVAILABLE and sandyFame < 2 then
+        -- don't meet fame requirements
+        player:startEvent(134)
+    elseif questIntroToTeamwork == QUEST_ACCEPTED then
+        -- start the event that asks them if they are ready to check
+        player:startEvent(129, 0, 1)
+    elseif questIntroToTeamwork == QUEST_COMPLETED and questIntermediateTeamwork == QUEST_AVAILABLE and sandyFame >= 3 and player:getMainLvl() >= 10 then
+        -- they've completed the first quest, the second quest is available, and they have thee required fame,
+        -- start event that starts the next quest
+        player:startEvent(133)
+    elseif questIntroToTeamwork == QUEST_COMPLETED and questIntermediateTeamwork == QUEST_AVAILABLE and sandyFame < 3 and player:getMainLvl() < 10 then
+        -- don't meet fame requirements
+        player:startEvent(132)
+    elseif questIntermediateTeamwork == QUEST_ACCEPTED then
+         -- start the event that asks them if they are ready to check
+        player:startEvent(129, 0, 2)
+    elseif questIntermediateTeamwork == QUEST_COMPLETED and questAdvancedTeamwork == QUEST_AVAILABLE and sandyFame >= 4 and player:getMainLvl() >= 10 then
+        -- they've completed the second quest, the third quest is available, and they have thee required fame,
+        -- start event that starts the next quest
+        player:startEvent(131)
+    elseif questIntermediateTeamwork == QUEST_COMPLETED and questAdvancedTeamwork == QUEST_AVAILABLE and sandyFame < 4 and player:getMainLvl() < 10 then
+        -- don't meet fame requirements
+        player:startEvent(130)
+    elseif questAdvancedTeamwork == QUEST_ACCEPTED then
+         -- start the event that asks them if they are ready to check
+        player:startEvent(129, 0, 3)
+    elseif questAdvancedTeamwork == QUEST_COMPLETED then
+        player:startEvent(130)
     else
-        player:startEvent(136) -- Default - before quests
+        player:startEvent(136)
     end
 end
 
 function onEventUpdate(player, csid, option)
+    -- csid 129 happens for both quests
+    if csid == 129 then
```

Comment:
Is there an outgoing event option that the player selects to confirm that they're ready for the check?

Reply:
Believe it or not, both options are 0. The 'automated message' packet is what differentiates the response.